### PR TITLE
Named Mutators and MultiMutator API change

### PR DESCRIPTION
**Pull request by domenukk**
This adds the `Named` attribute to all mutators, which may be more useful down the road (vs. taking names away as #1379 suggests)

cc @addisoncrump

### CODE REVIEW

**Review by addisoncrump — APPROVED** [review]

File: libafl/src/mutators/token_mutations.rs [hunk 1]
```diff
@@ -1090,37 +1090,58 @@ impl AFLppRedQueen {
...
+    ) -> Result<MutationResult, Error> {
+        let mut result = self.multi_mutate(state, input, stage_idx, Some(1))?;
+        if let Some(first_input) = result.pop() {
+            *input = first_input;
+            Ok(MutationResult::Mutated)
+        } else {
+            Ok(MutationResult::Skipped)
+        }
+    }
+}
```
Comment (addisoncrump) [inline libafl/src/mutators/token_mutations.rs:1112]: New trait implementor? :p

**Comment by domenukk** [comment]
Also @tokatoka right now all `Afl` names are `Afl`, Redqueen uses `AFLpp`, which is odd. How about just renaming it to `RedQueen` (no Afl) and referencing the original implementation in the docs?

File: libafl/src/mutators/token_mutations.rs (hunk 1 above)
Comment (addisoncrump) [reply libafl/src/mutators/token_mutations.rs:1112]: Ah I see, it's a required trait now. Nevermind.

Comment (domenukk) [reply libafl/src/mutators/token_mutations.rs:1112]: Maybe it shouldn't be(?) But I feel like it can be, so..

**Comment by tokatoka** [comment]
>  How about just renaming it to RedQueen (no Afl) and referencing the original implementation in the docs?

ok

File: libafl/src/mutators/token_mutations.rs [hunk 2]
```diff
@@ -1090,37 +1090,58 @@ impl AFLppRedQueen {
...
+{
+    fn mutate(
+        &mut self,
+        state: &mut S,
+        input: &mut I,
+        stage_idx: i32,
+    ) -> Result<MutationResult, Error> {
+        let mut result = self.multi_mutate(state, input, stage_idx, Some(1))?;
+        if let Some(first_input) = result.pop() {
+            *input = first_input;
```
Comment (tokatoka) [inline libafl/src/mutators/token_mutations.rs:1106]: nono we return.a vector here
we want the entire `result` vector

Comment (tokatoka) [reply libafl/src/mutators/token_mutations.rs:1106]: if result.len() > 0 => mutated else skipped

Comment (domenukk) [reply libafl/src/mutators/token_mutations.rs:1106]: Yes and a single mutate returns the first element of the vector

Comment (domenukk) [reply libafl/src/mutators/token_mutations.rs:1106]: But I can remove the single mutate if you don't think that makes sense

Comment (domenukk) [reply libafl/src/mutators/token_mutations.rs]: Removed

File: libafl/src/mutators/token_mutations.rs (hunk 1 above)
Comment (domenukk) [reply libafl/src/mutators/token_mutations.rs]: Removed it

File: libafl/src/mutators/mod.rs [hunk 3]
```diff
@@ -102,18 +103,20 @@ pub trait Mutator<I, S> {
...
+    /// Mutate a given input up to `max_count` times,
+    /// or as many times as appropriate, if no `max_count` is given
+    fn multi_mutate(
         &mut self,
         state: &mut S,
         input: &I,
-        vec: &mut Vec<I>,
         stage_idx: i32,
-    ) -> Result<MutationResult, Error>;
+        max_count: Option<usize>,
```
Comment (domenukk) [inline libafl/src/mutators/mod.rs:114]: @tokatoka should we have a `max_count` here or leave it out? Could make sense if we have a vectorized mutator for example

**Comment by domenukk** [comment]
No space left on device :/

**Pull request merged**